content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,412 @@ preview: true
+
+The following assumes you have upgraded to 2.26.0 and [enabled the preview feature flag](#enable-the-referentialactions-preview-feature).
+
+### Using Introspection
+
+If you [Introspect](../introspection) your database, the referential actions configured at the database level will be reflected in your Prisma Schema. If you have been using Prisma Migrate or `prisma db push` to manage the database schema, these are likely to be the [<=2.25.0 default values](#the-2250-default-referential-actions). 
+
+When you run an Introspection, Prisma compares all the foreign keys in the database with the schema, if the SQL statements `ON DELETE` and `ON UPDATE` do **not** match the default values, they will be explicitly set in the schema file.
+
+After introspecting, you can review the non-default clauses in your schema. The most important clause to review is `onDelete`, which defaults to `Cascade` in 2.25.0 and earlier.
+
+<Admonition type="alert">
+
+If you are using either the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods, **[cascading deletes](#how-to-use-cascading-deletes) will now be performed** as the `referentialAction` preview feature **removed the safety net in Prisma Client that previously prevented cascading deletes at runtime**. Be sure to check your code and make any adjustments accordingly.
+
+</Admonition>
+
+Make sure you are happy with every case of `onDelete: Cascade` in your schema - if not, modify your schema and push/migrate, or modify the underlying database if you are using an introspection-only workflow.
+
+The following example would result in a cascading delete, if the `User` is deleted then all of their `Post`'s will be deleted too.
+
+#### A blog schema example
+
+```prisma highlight=4;add
+model Post {
+  id        Int     @id @default(autoincrement())
+  title     String
+  author    User    @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId  Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+### Using Migration
+
+When running a [Migration](../../prisma-migrate) (or the [`prisma db push`](../../prisma-migrate/db-push) command) the [new defaults](#referential-action-defaults) will be applied to your database.
+
+<Admonition type="info">
+
+Unlike when you run an Introspect for the first time, the new referential actions clause and property, will **not** automatically be addded to your prisma schema by the Prisma VSCode extension. 
+You will have to manually add them if you wish to use anything other than the new defaults.
+
+</Admonition>
+
+Using referential actions is optional and you are not required to add them to your prisma schema. If you choose to omit them, the [new defaults](#referential-action-defaults) will be used. 
```

Comment:
```suggestion
Using referential actions is optional and you are not required to add them to your Prisma schema. If you choose to omit them, the [new defaults](#referential-action-defaults) will be used. 
```

Reply:
"Using referential actions is optional and you are not required to add them to your Prisma schema. If you choose to omit them, the [new defaults](#referential-action-defaults) will be used."

Small tweak here - I would say _using_ referential actions isn't optional, because there are defaults, explicitly defining referential actions in the Prisma schema is definitely optional.  

"Explicitly defining referential actions in your Prisma schema is **optional**. If you do not explicitly define a referential action for a relation, Prisma uses the [new defaults](#referential-action-defaults)."

VERY superfluous comment that you can fling in the bin: This sentence - 'If you choose to omit them, the [new defaults](#referential-action-defaults) will be used.' is  passive tense ('new defaults will be used' rather than 'Prisma will use the new defaults'). Passive tense can be hard to digest because you ask yourself "will be used.. by who?". One amusing way to check is to add "..by monkeys" at the end of a sentence and see if still makes sense, in which case it's passive tense 🐒

Reply:
> VERY superfluous comment that you can fling in the bin: This sentence - 'If you choose to omit them, the new defaults will be used.' is passive tense ('new defaults will be used' rather than 'Prisma will use the new defaults'). Passive tense can be hard to digest because you ask yourself "will be used.. by who?". One amusing way to check is to add "..by monkeys" at the end of a sentence and see if still makes sense, in which case it's passive tense 🐒

This is brilliant! Thank you!